File: ProblemSet1 - C /mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // Prompt user for the pyramid's heigh
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    // Print a pyramid of that height
    for (int row = 0; row < height; row++)
    {
        // Print spaces
        for (int space = 0; space < height - row - 1; space++)
        {
            printf(" ");
        }
        // Print row of bricks
        for (int column = 0; column <= row; column++)
        {
            printf("#");
        }
        printf("\n");
    }
}

File: ProblemSet6 - Python/sentimental-mario-less/mario.py
from cs50 import get_int

def main():
    height = get_height("Height: ")
    half_pyramid(height)

# Prompt user for valid input
def get_height(prompt):
    while True:
        h = get_int(prompt)

        # Valid input must be between 1 and 8 inclusive
        if h > 0 and h < 9:
            return h

# Print half pyramid
def half_pyramid(rows):
    for row in range(1, rows + 1):
        print(" " * (rows - row) + "#" * row)

if __name__ == "__main__":
    main()
